### PR TITLE
Set ERROR state when a step returns StepResult.FAILURE

### PR DESCRIPTION
**Pull request by radix**
Pretty straightforward! Had to add an `UpdateGroupStatus` intent/performer.

### CODE REVIEW

**Comment by radix** [comment]
jenkins please run integration tests

File: otter/test/convergence/test_service.py [hunk 1]
```diff
@@ -759,11 +760,26 @@ def plan(*args, **kwargs):
                 TestStep(Effect(Constant((StepResult.SUCCESS, [])))),
             ])
 
+        def noop(i):
```
Comment (manishtomar) [inline otter/test/convergence/test_service.py]: Many intent performers need this. Would be nice to have it in utils?

**Comment by manishtomar** [comment]
LGTM modulo non-blocking nitpick above :)

**Comment by radix** [comment]
Thanks @manishtomar :)

**Pull request merged**